common/persistence/sql/sqlplugin/db_metrics.go

```diff
@@ -0,0 +1,86 @@
+package sqlplugin
+
+import (
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"go.temporal.io/server/common/log"
+	"go.temporal.io/server/common/metrics"
+)
+
+type DBMetricsReporter struct {
+	interval time.Duration
+	handle   *DatabaseHandle
+	metrics  metrics.Handler
+	quit     chan struct{}
+	wg       sync.WaitGroup
+
+	started int32
+	stopped int32
```

Comment:
Can use atomic bools here: https://pkg.go.dev/sync/atomic#Bool

Reply:
But also wondering why this would be started multiple times since it's only instantiated in the handle constructor and immediately started there.

Reply:
I switched it to `atomic.Bool` in 921be964f445d89a0f6ed1c9f35936bc7b524b17

Yeah, `Start` will probably only be called once, but `Stop` might be called multiple times, which I'm not sure about. Just keep it symmetric with the stop.

Before the guard check, the Go race detector complains about a race condition in WaitGroup. I'm adding these to resolve the race condition